dev/src/bulk-writer.ts

```diff
@@ -174,10 +174,16 @@ class BulkCommitBatch {
   /**
    * Resolves the individual operations in the batch with the results.
    */
-  processResults(results: BatchWriteResult[]): void {
-    results.map((result, index) => {
-      this.resultsMap.get(index)!.resolve(result);
-    });
+  processResults(results: BatchWriteResult[], error?: Error): void {
+    if (error === undefined) {
+      results.map((result, index) => {
```

Comment:
Can you use a `for` loop here as well? `map` creates a copy of the result that you are then throwing away.

Reply:
done.